### ISSUE TITLE
Generator can't read outer variable in fragment

### ISSUE THREAD

**Issue by westito** [heftapp/graphql_codegen#84]
Generator fails when I try to generate this graphql (the commented out field)

`Bad state: Failed to generate type.`

```graphql
schema {
    query: Query
}

type Competition {
    "Country code of the competition."
    countryCode: String
    "Relative URL for the country flag of the competition."
    flagUrl: String
    "Unique ID of the competition."
    id: Int!
    "Localized name."
    name(locale: Locale = en_US): String
}


type Query {
    competitions(
        "Filter competitions by date. If not provided, the default value is current date."
        date: DateTime,
        "If it's true, retrieve competitions with live matches only."
        live: Boolean = false
    ): [Competition!]!
}

type Team {
    "Number of goals on the current match."
    goals: Int
    "Unique ID of the team."
    id: Int!
    "Relative URL to the jersey image of current team."
    jerseyUrl: String
    "Lineup and bench players of the team on the current match."
    lineup: [Person!]!
    "Localized name."
    name(locale: Locale = en_US): String
    "Localized TLA name."
    tlaName(locale: Locale = en_US): String
}

enum Locale {
    da_DK
    de_DE
    el_GR
    en_US
    es_ES
    fr_FR
    it_IT
    nl_NL
    pl_PL
    pt_BR
    tr_TR
}

"A date-time string at UTC, such as 2019-12-03T09:54:33Z, compliant with the date-time format."
scalar DateTime

query MainPage($date: DateTime!, $locale: Locale!, $live: Boolean!) {
    competitions(date: $date, live: $live) {
        id
        flagUrl
        name(locale: $locale)

        ...basicTeamInfo
    }
}

fragment basicTeamInfo on Team {
    id
    goals
    #name(locale: $locale)
    jerseyUrl
}
```

**Comment by budde377** [comment]
I can see that you're spreading a fragment on `Team` on the type `Competition`. The `Competition` is not of type `Team`, I would guess that this is the issue.

**Comment by westito** [comment]
No. Sorry. I didn't want to copy the whole
```graphql
query MainPage($date: DateTime!, $locale: Locale!, $live: Boolean!) {
  competitions(date: $date, live: $live) {
    id
    flagUrl
    name(locale: $locale)

    counter: matchesCount {
      live
      total
    }

    matches {
      id
      state
      minute
      kickoff
      incident

      home {
        ...basicTeamInfo
      }

      away {
        ...basicTeamInfo
      }
    }
  }
}
```

The fragment are spread on Team.

**Comment by budde377** [comment]
Okay, I can't really reproduce this with a partial schema.

**Comment by westito** [comment]
Sorry, I had to ask for a permission to share the whole.
This is appropriate for you sharing by Dropbox?
https://www.dropbox.com/s/vbp0nbydpfr8qzs/schema.graphql?dl=1

**Comment by budde377** [comment]
@westito, thanks for reporting this and sharing your schema. I've found the bug and a fix is out in `0.8.1`.
